--- src/ffi.rs.orig	2016-05-29 13:12:59.639187785 +0100
+++ src/ffi.rs	2016-05-29 13:13:08.104810007 +0100
@@ -4,6 +4,12 @@
          non_camel_case_types,
          non_upper_case_globals,
          non_snake_case)]
+use c_types::fd_set;
+use c_types::in_addr;
+use c_types::sockaddr;
+use c_types::socklen_t;
+use libc::size_t;
+use libc::timeval;
 pub type ares_socklen_t = socklen_t;
 pub type __cares_rule_02__ = [::std::os::raw::c_char; 1usize];
 pub type __cares_rule_03__ = [::std::os::raw::c_char; 1usize];
@@ -111,7 +117,6 @@
 }
 #[repr(C)]
 #[derive(Copy, Clone)]
-#[derive(Debug)]
 pub struct ares_addrttl {
     pub ipaddr: in_addr,
     pub ttl: ::std::os::raw::c_int,
